src/network/networkd-link.c

```diff
@@ -2714,8 +2765,18 @@ int link_update(Link *link, sd_netlink_message *m) {
         }
 
         r = sd_netlink_message_read_strv(m, IFLA_PROP_LIST, IFLA_ALT_IFNAME, &s);
-        if (r >= 0)
+        if (r >= 0) {
+                char **n;
+
+                STRV_FOREACH(n, link->alternative_names)
+                        hashmap_remove_value(link->manager->links_by_name, *n, link);
+
+                r = link_update_by_name(link);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add alternative name '%s' on network interface '%s', ignoring: %m", *n, link->ifname);
```

Comment:
you are not actually ignoring here. I think ignoring is good, but please downgrade to `log_warning_errno()`, and drop the `return`